### ISSUE TITLE
Go To Definition and Go To Implementation should be ref assembly aware.

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn#21217]
When using the new 'reference assembly' feature of the compiler, the IDE takes you to the metadata view when using 'Go To Definition' and displays a dialog indicating "The symbol has no implementations" when using 'Go To Implementation'

I would expect the IDE to be 'aware' of this feature and to still take you to the source file in the solution.

**Comment by CyrusNajmabadi** [comment]
This is unrelated to the actual features, and is more about what the project-system/workspace do here.  It sounds like they're attaching ref-assemblies as actual assemblies, instead of as P2P references.  Tagging @jasonmalinowski

**Comment by Pilchie** [comment]
@tannergooding do you have https://github.com/dotnet/roslyn/pull/20833  @sharwell already added some support there.

**Comment by tannergooding** [comment]
@Pilchie, this was on vsuml.26724.4002

**Comment by sharwell** [comment]
@tannergooding I just tested this with 15.3 Preview 6 and both Go To Definition and Go To Implementation were working as expected. I enabled the feature like this for testing:

```xml
<PropertyGroup>
  <ProduceReferenceAssembly>true</ProduceReferenceAssembly>
  <CompileUsingReferenceAssemblies>true</CompileUsingReferenceAssemblies>
</PropertyGroup>
```

**Comment by tannergooding** [comment]
Ok. I confirmed this is fixed in 15.3 Preview 6. It seems VSUML doesn't have the change yet.
